### PR TITLE
replace babel-loader with swc-loader

### LINKED ISSUE #28779: replace babel-loader for esbuild-loader
esbuild-loader seems to support now auto detection of the file it's parsing, in the same way babel-loader doe (added a few weeks ago: https://github.com/esbuild-kit/esbuild-loader/releases).

So before you had to do

```
{
  test: /\.tsx?$/,
  exclude: /node_modules|cljs/,
  use: [{ loader: "esbuild-loader", options: { loader: 'tsx'} }],
},
{
  test: /\.jsx?$/,
  exclude: /node_modules|cljs/,
  use: [{ loader: "esbuild-loader", options: { loader: 'jsx'} }],
},
```

and now you should be able to do
```
{
  test: /\.[jt]sx?$/,
  exclude: /node_modules|cljs/,
  loader: "esbuild-loader",
},
```

but this seems to break with some .js files we have that should be .jsx (both in the frontend and in static-viz webpack configs).

I tested this and it seems to shed 50% of build time, but it breaks the FE, so I'm guessing something else has to be fixed for this to work.

Oh, and I also forgot about replacing babel-loader for using emotion, which is not very clear how that can be fixed

### LINKED ISSUE #37577: Migrate FE webpack from babel-loader to the SWC loader
The current webpack build takes about **70-80** seconds on CI, and this time can be almost halved if we stop using babel-loader.
The main problem is that we use a plugin for Emotion, which also needs to be migrated, and in the long term, we should consider moving away from Emotion.

Here is [PoC PR](https://github.com/metabase/metabase/pull/37054) of the SWC loader you can use as the reference.
The only known issue is that we need to figure out why the Cypress tests are not running on CI. Locally, everything works fine.

```[tasklist]
### Tasks
- [ ] Investigate the issue with Cypress 
- [ ] https://github.com/metabase/metabase/issues/45905
- [ ] Remove all babel references 
- [ ] https://github.com/metabase/metabase/issues/45906
```

### PR DESCRIPTION
**Pull request by uladzimirdev**
Closes https://github.com/metabase/metabase/issues/28779
Closes https://github.com/metabase/metabase/issues/37577

[esbuild PoC](https://github.com/metabase/metabase/pull/45845/files)

### Description

Replace babel-loader with swc-loader to speed up production (and sometimes development) webpack build.

Notes: 
- babel is kept to transpile fonts file as swc emotion plugin doesn't handle our case correctly. I didn't find a root cause of wrong behaviour, but the safe past is to keep using babel + emotion plugin for one file and all other can be transpiled with swc
- react fast-refresh now handled by swc
- swc-loader can't read browserslist config from package.json, so we had to manually specify supported browsers in webpack

### How to verify

- CI is green
- `yarn dev` works as before
- `yarn build-release` produces more or less similar size js and css files

#### Before
image

#### After

image

### CODE REVIEW

**Comment by iethree** [comment]
Looks good to me! 🥳  🚀

**Review by iethree — APPROVED** [review]

File: webpack.config.js [hunk 1]
```diff
@@ -102,9 +133,15 @@ const config = (module.exports = {
   module: {
     rules: [
       {
-        test: /\.(tsx?|jsx?)$/,
+        // swc breaks styles for the whole app if we proceed this file
```
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply webpack.config.js:136]: I'll fix it in one of followups

File: frontend/src/metabase-lib/v1/queries/structured/Filter.ts [hunk 2]
```diff
@@ -354,7 +354,7 @@ export default class Filter extends MBQLClause {
     return this.set(setFilterOptions(this, options));
   }
 
-  formattedArguments(maxDisplayValues?: number = 1) {
+  formattedArguments(maxDisplayValues: number = 1) {
```
Comment (nemanjaglumac) [inline frontend/src/metabase-lib/v1/queries/structured/Filter.ts:357]: How was this related to webpack -> SWC migration?

Comment (uladzimirdev) [reply frontend/src/metabase-lib/v1/queries/structured/Filter.ts:357]: @nemanjaglumac @WiNloSt it's not correct typescript syntax, so swc complained about it 😄

Comment (nemanjaglumac) [reply frontend/src/metabase-lib/v1/queries/structured/Filter.ts:357]: Thanks

File: webpack.config.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (uladzimirdev) [reply webpack.config.js:136]: https://github.com/metabase/metabase/pull/45902/files#diff-1fb26bc12ac780c7ad7325730ed09fc4c2c3d757c276c3dacc44bfe20faf166fR138

**Pull request merged**